From fc05d31d53f521b13bc6c99f8f071e4835303cf2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:00 -0400
Subject: [usb] serial: export usb_serial_generic_write_start

Message-id: <1410887463-20674-49-git-send-email-dzickus@redhat.com>
Patchwork-id: 95138
O-Subject: [RHEL7 PATCH 048/591] USB: serial: export usb_serial_generic_write_start
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 706cd17e8559c96dc883ba692c931f1ef31fbc5c
Author: Johan Hovold <jhovold@gmail.com>
Date:   Wed Oct 9 17:01:12 2013 +0200

    USB: serial: export usb_serial_generic_write_start

    Export usb_serial_generic_write_start which is needed when implementing
    a custom resume function while still relying on the generic write
    implementation.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/generic.c b/drivers/usb/serial/generic.c
index 31f7829..2b01ec8 100644
--- a/drivers/usb/serial/generic.c
+++ b/drivers/usb/serial/generic.c
@@ -129,7 +129,7 @@ int usb_serial_generic_prepare_write_buffer(struct usb_serial_port *port,
  *
  * Return: Zero on success or if busy, otherwise a negative errno value.
  */
-static int usb_serial_generic_write_start(struct usb_serial_port *port,
+int usb_serial_generic_write_start(struct usb_serial_port *port,
        gfp_t mem_flags)
 {
  struct urb *urb;
@@ -184,6 +184,7 @@ retry:
 
  return 0;
 }
+EXPORT_SYMBOL_GPL(usb_serial_generic_write_start);
 
 /**
  * usb_serial_generic_write - generic write function
diff --git a/include/linux/usb/serial.h b/include/linux/usb/serial.h
index d528b80..704a1ab 100644
--- a/include/linux/usb/serial.h
+++ b/include/linux/usb/serial.h
@@ -320,6 +320,8 @@ extern struct usb_serial_port *usb_serial_port_get_by_minor(unsigned int minor);
 extern void usb_serial_put(struct usb_serial *serial);
 extern int usb_serial_generic_open(struct tty_struct *tty,
  struct usb_serial_port *port);
+extern int usb_serial_generic_write_start(struct usb_serial_port *port,
+       gfp_t mem_flags);
 extern int usb_serial_generic_write(struct tty_struct *tty,
  struct usb_serial_port *port, const unsigned char *buf, int count);
 extern void usb_serial_generic_close(struct usb_serial_port *port);
-- 
1.7.1